core/src/main/java/org/mule/util/ClassUtils.java

```diff
@@ -1065,4 +1067,70 @@ public static URL getClassPathRoot(Class clazz)
             return annotation.getClass();
         }
     }
+
+    /**
+     * Checks that {@code value} is an instance of {@code type}. The value
+     * that this method adds over something like {@link Class#isInstance(Object)}
+     * is that it also considers the case in which {@code type} and {@code value}
+     * are evaluate by {@link #isWrapperAndPrimitivePair(Class, Class)} as {@code true}
+     *
+     * @param type  the {@link Class} you want to check the {@code value against}
+     * @param value an instance you want to verify is instance of {@code type}
+     * @param <T>   the generic type of {@code type}
+     * @return {@code true} if {@code value} is an instance of {@code type} or if they are a wrapper-primitive pair.
+     *          {@code false} otherwise
+     */
+    public static <T> boolean isInstance(Class<T> type, Object value) {
```

Comment:
i think the current order is more intuitive since this is kinda replacing type.isInstance(value)